src/workflows/airqo_etl_utils/tests/calibration/test_pipeline.py

```
@@ -119,40 +124,41 @@ def test_run_captures_error_per_country_without_crashing(self, mock_extract):
-        df, features = pipeline._prepare_training_data(
-            lcs_other,
-            bam_raw_df,
-            COUNTRY_CONFIG["uganda"],
-        )
-        # Should not raise; falls back to all LCS rows
-        assert (
-            not df.empty or df.empty
-        )  # either outcome is valid here; no exception raised
+        # Falls back to all devices when primary is absent — should not raise
+        try:
+            df, features = pipeline._prepare_training_data(
```

Ruff (0.15.12)

[warning] Unpacked variable `df` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)